File: src/python/square_area.py
import sys

"""
Módulo para el cálculo del área de figuras geométricas.
Ecuación principal: A = a^2 (Área = base al cuadrado).
"""

def calculate_square_area(input_value):
    """
    Realiza el cálculo del área de un cuadrado.

    Args:
        input_value (str|float): El valor de la base 'a' ingresado por el usuario.

    Returns:
        float: El área 'A' calculada.

    Raises:
        ValueError: Si la entrada es inválida o el valor es negativo.
    """
    try:
        # Convertir la entrada a flotante para permitir decimales
        a = float(input_value)
        
        # Validación geométrica: la base no puede ser negativa
        if a < 0:
            raise ValueError("La base 'a' no puede ser negativa en un cuadrado.")
            
        # Implementación de la ecuación fundamental: A = a^2
        A = a ** 2
        return A
        
    except ValueError as e:
        if "negativa" in str(e):
            raise e
        raise ValueError("Por favor, ingrese un número válido para la base 'a'.")

def process_area(input_value):
    """
    Llama al cálculo del área y maneja la salida.
    """
    try:
        A = calculate_square_area(input_value)
        print(f"Resultado: El área (A) del cuadrado con base (a = {input_value}) es: {A}")
    except ValueError as e:
        print(f"Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    """
    Punto de entrada principal del script.
    Soporta argumentos de línea de comandos o entrada interactiva.
    """
    if len(sys.argv) > 1:
        # Uso mediante argumentos: python square_area.py 5
        process_area(sys.argv[1])
    else:
        # Uso interactivo: solicita el dato al usuario
        try:
            base_input = input("Ingrese el valor de la base (a): ")
            process_area(base_input)
        except (EOFError, KeyboardInterrupt):
            sys.exit(0)

File: src/cpp/square_area.cpp
/**
 * @file square_area.cpp
 * @author German
 * @brief Programa CLI para calcular el área de un cuadrado usando la fórmula A = a^2.
 * @version 1.1.0
 * @date 2026-03-07
 */

#include <iostream>
#include <cmath>
#include <string>
#include <cstdlib>
#include <iomanip>

/**
 * @brief Calcula el área (A) a partir de la base (a) y muestra el resultado por consola.
 * 
 * Implementa la validación de seguridad geométrica para evitar valores negativos.
 * 
 * @param a Valor decimal que representa la base del cuadrado.
 * @return void
 */
void print_area(double a) {
    // Validación: Una base negativa no es válida para figuras geométricas reales.
    if (a < 0) {
        std::cerr << "Error: La base 'a' de un cuadrado no puede ser negativa." << std::endl;
        std::exit(EXIT_FAILURE);
    }

    /**
     * @details Aplicación de la ecuación fundamental del área:
     * A = a^2
     */
    double A = std::pow(a, 2);

    // Salida estándar con descripción clara de las variables de la ecuación.
    std::cout << "Resultado: El área (A) del cuadrado con base (a = " << a << ") es: " << A << std::endl;
}

/**
 * @brief Función principal (punto de entrada).
 * 
 * Maneja la captura de datos desde argumentos de terminal (argc/argv) o mediante
 * flujo de entrada estándar (std::cin).
 * 
 * @param argc Conteo de argumentos.
 * @param argv Lista de argumentos de texto.
 * @return int Estado de salida del programa (EXIT_SUCCESS o EXIT_FAILURE).
 */
int main(int argc, char* argv[]) {
    double base;

    if (argc > 1) {
        /**
         * Manejo de argumentos CLI: ./square_area 10.5
         * std::atof convierte cadena a flotante de doble precisión.
         */
        base = std::atof(argv[1]);
        
        // Validación técnica: El valor 0.0 podría ser un error de conversión si el input es texto.
        if (base == 0 && argv[1][0] != '0') {
            std::cerr << "Error: Por favor, ingrese un número válido para la base 'a'." << std::endl;
            return EXIT_FAILURE;
        }
    } else {
        /**
         * Manejo interactivo: Solicita el dato si no se pasó como argumento.
         */
        std::cout << "Ingrese el valor de la base (a): ";
        if (!(std::cin >> base)) {
            std::cerr << "Error: Entrada inválida. Se esperaba un valor numérico." << std::endl;
            return EXIT_FAILURE;
        }
    }

    // Ejecución del cálculo y salida
    print_area(base);

    return EXIT_SUCCESS;
}
